src/briefcase/integrations/virtual_environment.py

```diff
@@ -0,0 +1,65 @@
+import subprocess
+import sys
+from pathlib import Path
+
+from briefcase.config import AppConfig
+from briefcase.console import Console
+from briefcase.exceptions import BriefcaseCommandError
+
+
+class VenvEnvironment:
+    def __init__(self, tools, console: Console, base_path: Path, app: AppConfig):
+        self.tools = tools
+        self.console = console
+        self.app = app
+        self.venv_path = base_path / ".briefcase" / app.app_name / "venv"
+        self.pyvenv_cfg = self.venv_path / "pyvenv.cfg"
+
+    def __enter__(self):
+        self.console.info(
+            f"Looking for isolated virtual environment for {self.app.app_name}."
+        )
+
+        if self.pyvenv_cfg.exists():
+            self.console.info("Found existing virtual environment. Skipping creation.")
+        else:
+            self.console.info("No virtual environment found. Creating...")
```

Comment:
If there's a task that will take an indeterminate amount of time, use `console.wait_bar` - that adds a "spinner", and handles printing the `done` at the end as well.